include/internal/catch_test_case_registry_impl.cpp

```diff
@@ -15,10 +15,34 @@
 #include "catch_string_manip.h"
 #include "catch_test_case_info.h"
 
+#include <random>
 #include <sstream>
 
 namespace Catch {
 
+    struct CompareTestsUsingRandom {
+        explicit CompareTestsUsingRandom( Catch::SimplePcg32& rng ) {
+            std::uniform_int_distribution<uint64_t> dist;
+            basis = dist( rng );
+        }
+
+        uint64_t basis;
+
+        uint64_t fnv_1a_hash( std::string const& s ) const {
+            static constexpr uint64_t prime = 1099511628211;
```

Comment:
I am not sure why the compiler doesn't warn on this, but AFAIK the constant will be truncated. To generate `uint64_t` constants, you need `ULL` suffix.

Reply:
No.  The compiler will automatically pick the smallest integer type sufficient to represent the constant.  See e.g. N3337 [lex.icon] paragraph 2.

Reply:
[Oh right](https://eel.is/c++draft/lex.icon#3). Nevermind.